wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala

```diff
@@ -96,59 +96,78 @@ private[wallet] trait UtxoHandling extends WalletLogger {
       spendingInfoDbs: Vector[SpendingInfoDb]): Future[
     Vector[SpendingInfoDb]] = {
 
-    val byBlock = spendingInfoDbs.groupBy(_.blockHash)
-
-    val toUpdateFs = byBlock.map {
-      case (Some(blockHash), txos) =>
-        chainQueryApi.getNumberOfConfirmations(blockHash).map {
-          case None =>
-            logger.warn(
-              s"Given txos exist in block (${blockHash.hex}) that we do not have or that has been reorged! $txos")
-            Vector.empty
-          case Some(confs) =>
-            txos.flatMap { txo =>
-              txo.state match {
-                case TxoState.ImmatureCoinbase =>
-                  if (confs > Consensus.coinbaseMaturity) {
+    val txIds =
```

Comment:
So this method is really blowing up in size and complexity. Usually that is a sign that we need to break it up into smaller methods that are testable. 

I don't see any unit tests that directly test this logic, and if I understand our initial bug, it originates from here. 

I think we should change the scope of this method to be `private[wallet]` and add unit tests for various cases like CPFP which i mention below.